locator/abstract_replication_strategy.hh

```diff
@@ -399,6 +401,15 @@ public:
         return _stopped;
     }
 
+    // Note that func is called with a const effective_replication_map&
+    // If it yields, it must keep erm.shared_from_this() to extend its lifetime
+    template <typename Func>
+    requires std::invocable<Func, const effective_replication_map&>
+    future<> do_for_each(Func func) const noexcept {
```

Comment:
Can simplify to `future<> do_for_each(std::invocable<const effective_replication_map&> to func)`

Reply:
Didn't work for me... Please see https://github.com/scylladb/scylla/pull/9703#discussion_r758967952

Reply:
Fixed in v4